src/Deserialization/OperandFactory.cs

```diff
@@ -0,0 +1,143 @@
+﻿using System;
+using System.Globalization;
+using System.Linq;
+using System.Numerics;
+using LLVMSharp.Interop;
+using Symbolica.Abstraction;
+using Symbolica.Deserialization.Extensions;
+using Symbolica.Expression;
+using Symbolica.Representation;
+using Symbolica.Representation.Operands;
+
+namespace Symbolica.Deserialization
+{
+    internal sealed class OperandFactory : IOperandFactory
+    {
+        private readonly IIdFactory _idFactory;
+        private readonly LLVMTargetDataRef _targetData;
+        private readonly IUnsafeContext _unsafeContext;
+
+        public OperandFactory(LLVMTargetDataRef targetData, IIdFactory idFactory, IUnsafeContext unsafeContext)
+        {
+            _targetData = targetData;
+            _idFactory = idFactory;
+            _unsafeContext = unsafeContext;
+        }
+
+        public IOperand Create(LLVMValueRef operand, IInstructionFactory instructionFactory)
+        {
+            return operand.Kind switch
+            {
+                LLVMValueKind.LLVMArgumentValueKind =>
+                    new Argument(int.Parse(operand.GetValue()[1..])),
```

Comment:
Here is a use of the string dump nonsense I mention above.

Reply:
If I'm not mistaken isn't `GetValue` an extension we've added? In that case, would it make sense to drop the leading `'%'` there?

Reply:
`GetValue` is used to get the full string raw "value of a value", and only in the argument case do we want to drop the first character. For getting integer and float scalars (which are the other 2 uses) we need the whole thing.